### PR TITLE
Rsyslog

### PR DESCRIPTION
**Pull request by oleksdiam**
Rsyslog server has been configured to receive logs from remote clients & sort them by hostnames.
Rsyslog clients collect logs from node applications & send them through TCP port to server along with sysem logs. Use queues to avoid loss of data. Execution of rsyslog::client defined type have to be inserted in appropiated profile. For example:
  $log_proto     = 'tcp'
  $log_port      = '601'
  $log_serv      = '192.168.56.10'
  $application1 = {
    log_name => '/path/to/logs/*.log',
    app_name => 'appname_for_tag',
    severity => 'info',
  }
  $application2 = {
    log_name => '/path/to/logs/*.log',
    app_name => 'appname_for_tag',
    severity => 'info',
  }
  $apps   = [$application1, $application2]

  rsyslog::client { 'app' :
    log_proto    => $log_proto,
    log_port     => $log_port,
    log_serv     => $log_serv,
    apps         => $apps,
  }

### CODE REVIEW

**Review by turnopil — COMMENTED** [review]
Does jenkins have 6.7 version? $jenkins_version = '6.7.2'

File: modules/postgres/manifests/init.pp [hunk 1]
```diff
@@ -119,27 +109,21 @@
...
-    # notify      => Service['firewalld'],
-  }
-
-  # ->
-  # service { 'firewalld':
-  #   ensure     => running,
-  #   enable     => true,
-  #   hasrestart => true,
-  #   subscribe  => Exec['firewall-port'],
+  # if $setfirewall != 'off' {
```
Comment (ipleten) [inline modules/postgres/manifests/init.pp]: Each time you're trying to push code which is commented out one a kitten dies.

**Review by ipleten — COMMENTED** [review]
Please re-work code as  we have discussed today.

File: modules/profile/manifests/jenkins.pp [hunk 2]
```diff
@@ -1,4 +1,38 @@
...
+
+  base::ssh_user { $ssh_user:
+    ssh_user     => $ssh_user,
+    ssh_group    => $ssh_group,
+    ssh_password => $ssh_password,
+  }
+  firewall::openport {'jenkins':
+    dports       => $dports,
+  }
+  rsyslog::client { 'app' :
```
Comment (ipleten) [inline modules/profile/manifests/jenkins.pp]: is shouldn't be "app" it should be "jenkins"

I bet log_proto and log_port are the same across all roles. Use defaults!

File: modules/profile/manifests/rsyslog.pp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    app_name => 'sonar',
+    severity => 'info',
+  }
+  $apps   = [$sonar]
+
+  rsyslog::client { 'app' :
+    log_proto    => $log_proto,
+    log_port     => $log_port,
+    log_serv     => $log_serv,
+    apps         => $apps,
```
Comment (ipleten) [inline modules/profile/manifests/rsyslog.pp]: sonar = something
apps = sonar
app = apps..
%)
why don't just apps=something?

File: modules/profile/manifests/tomcat.pp [hunk 4]
```diff
@@ -17,6 +31,14 @@
   firewall::openport {'tomcat':
     dports          => $dports,
   }
+  
+  rsyslog::client { 'app' :
+    log_proto    => $log_proto,
+    log_port     => $log_port,
+    log_serv     => $log_serv,
+    apps         => $apps,
+  }
```
Comment (ipleten) [inline modules/profile/manifests/tomcat.pp]: oh... it should be
```
 rsyslog::client { 'tomcat' :
      log_serv     => $log_serv,
      log_name => '/var/log/tomcat/*.log',
      app_name => 'tomcat',
      severity => 'info'
 }

```
and 
```
rsyslog::client { 'httpd' :
      log_serv     => $log_serv,
      log_name => '/var/log/httpd/*.log',
      app_name => 'httpd',
      severity => 'info'
 }
```

File: modules/sonarqube/manifests/init.pp [hunk 5]
```diff
@@ -160,24 +160,24 @@
...
+  #     notify  => Service['firewalld'],
+  #   }
+  #   ->
+  #   service { 'firewalld':
+  #     ensure     => running,
+  #     enable     => true,
+  #     hasrestart => true,
+  #   }    
+  # }
+
```
Comment (ipleten) [inline modules/sonarqube/manifests/init.pp]: -1 kitten.

**Review by ipleten — CHANGES_REQUESTED** [review]
But some reason i can't see a code of rsyslog::client

File: modules/profile/manifests/rsyslog.pp (hunk 3 above)
Comment (oleksdiam) [reply modules/profile/manifests/rsyslog.pp]: Для уніфікації коду:
в загальному випадку apps - це список хешів
тому заявляю хеші, заганяю їх у список, а список передаю в defined type

File: modules/profile/manifests/tomcat.pp (hunk 4 above)
Comment (ipleten) [reply modules/profile/manifests/tomcat.pp]: I want to have something like that ^^^

Comment (ipleten) [reply modules/profile/manifests/tomcat.pp]: and in rsyslog::client you need to create file for each app in  /etc/rsyslog.d/
e.g:
/etc/rsyslog.d/httpd.conf
/etc/rsyslog.d/tomcat.conf

**Pull request closed without merging**